synapse/events/snapshot.py

```diff
@@ -131,8 +125,7 @@ class EventContext(UnpersistedEventContextBase):
     _state_group: Optional[int] = None
     state_group_before_event: Optional[int] = None
     _state_delta_due_to_event: Optional[StateMap[str]] = None
-    prev_group: Optional[int] = None
-    delta_ids: Optional[StateMap[str]] = None
+    state_group_deltas: Optional[Dict[Tuple[int, int], StateMap[str]]] = None
```

Comment:
I am left wondering if this should also be a `Dict` that's just empty sometimes? 🤷

Reply:
I also wondered about that - in the interest of reducing `None`s I can go ahead an do that.

Reply:
I don't think it is a big deal either way, was just a thought 🤷

Reply:
(It likely means you can skip some checks that it is none and just attempt to pull things out of it though?)